### PR TITLE
Establishes linting for TypeScript in GH Actions

### LINKED ISSUE #18: Establish a linting standard
We should have automated linting checks for our repo.
* See https://github.com/marketplace/actions/lint-action . I suspect I don't have privileges to add this.
* I recommend "Prettier" for .js linting.
* Is there a `clarity` linter yet?

Let's discuss @labs3/daoos :)

### PR DESCRIPTION
**Pull request by asteriabtc**
### plans

- [x] add GitHub Action linting for Typescript

will close #18

### CODE REVIEW

**Review by cryptopanter — APPROVED** [review]

**Comment by cryptopanter** [comment]
Is it a free service from Github?... And what lang does ot support? Javascript?...

**Comment by JMKJR** [comment]
> Is it a free service from Github?... And what lang does ot support? Javascript?...

@cryptopanter Github actions is a free ci/cd tool that will execute some server-side job on the occurrence of some event (eg. PR, push). So we can use these actions to do things like check if our code is properly linted or complies with Typescript type checking.

**Comment by JMKJR** [comment]
PR #37 Was merged into this one to complete the task of adding github actions. This PR is now ready for review @asteriabtc. Adding a Github action for custom linting of clarity contracts should be made into a separate issue.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,2 @@
-
-.DS_Store
+node_modules
+.DS_Store
```
Comment (JMKJR) [inline .gitignore:2]:
```suggestion
**/.DS_Store
```
If you want to include this in our `.gitignore`, let's make sure we also ignore .DS_Store files that could appear in subdirectories.

**Review by JMKJR — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (JMKJR) [reply .gitignore:2]: Or add `.DS_Store` to the `.gitignore` in our `packages/contracts` folder. It's already in the `.gitignore` in `packages/ui`

**Review by JMKJR — CHANGES_REQUESTED** [review]

**Review by JMKJR — APPROVED** [review]

**Comment by asteriabtc** [comment]
@JMKJR gonna force merge it now.

**Pull request merged**